pkg/db/db_client/db_client.go

```diff
@@ -109,13 +111,14 @@ func NewDbClient(ctx context.Context, connectionString string, onConnectionCallb
 	return client, nil
 }
 
-func (c *DbClient) loadServerSettings(ctx context.Context) error {
-	conn, _, err := c.GetDatabaseConnectionWithRetries(ctx)
-	if err != nil {
-		return err
+func (c *DbClient) closePools(ctx context.Context) {
```

Comment:
ctx not needed

rename to `closePool`
would pool ever be nil?

Reply:
dbclient creation fails if pool creation fails, so we can never have a db client without a pool, right? 

So null check is not necessary?